### PR TITLE
Twig50

### PR DESCRIPTION
**Pull request by nodatk**
Added my handwired keyboard.

## Description

Added my handwired keyboard twig50, with a default keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

none

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/twig/twig50/config.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/twig/twig50/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/handwired/twig/twig50/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,79 @@
+#include "twig50.h"
```
Comment (drashna) [inline keyboards/handwired/twig/twig50/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/handwired/twig/twig50/rules.mk [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+AUDIO_ENABLE = yes
+RGBLIGHT_ENABLE = no
+#LAYOUTS = diag_4x13
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
+NO_SUSPEND_POWER_DOWN = yes
+UNICODEMAP_ENABLE = no
```
Comment (drashna) [inline keyboards/handwired/twig/twig50/rules.mk:27]:
```suggestion
```

File: keyboards/handwired/twig/twig50/twig50.c [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#include "twig50.h"
```
Comment (drashna) [inline keyboards/handwired/twig/twig50/twig50.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/handwired/twig/twig50/twig50.c [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+
+void matrix_init_kb(void) {
+  matrix_init_user();
+}
```
Comment (drashna) [inline keyboards/handwired/twig/twig50/twig50.c:5]:
```suggestion
```

File: keyboards/handwired/twig/twig50/twig50.h [hunk 6]
```diff
@@ -0,0 +1,18 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/twig/twig50/twig50.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/twig/twig50/README.md [hunk 7]
```diff
@@ -0,0 +1,4 @@
+Handwired 50 key with Proton C.
```
Comment (zvecr) [inline keyboards/handwired/twig/twig50/README.md:1]: Should follow the [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md), and preferably be lowercase.

File: keyboards/handwired/twig/twig50/config.h (hunk 1 above)
Comment (nodatk) [reply keyboards/handwired/twig/twig50/config.h:4]: Thank you.
Added the comment.

**Comment by nodatk** [comment]
Thank you for your review and comments.
I fixed them, please check agin.

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/twig/twig50/rules.mk [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+# Build Options
+BACKLIGHT_ENABLE = no
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # USB Nkey Rollover
+AUDIO_ENABLE = yes
+RGBLIGHT_ENABLE = no
```
Comment (fauxpark) [inline keyboards/handwired/twig/twig50/rules.mk:17]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = yes          # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by nodatk** [comment]
@fauxpark 
Thank you for your review.
I fixed them, please check agin.

File: keyboards/handwired/twig/twig50/rules.mk [hunk 9]
```diff
@@ -0,0 +1,18 @@
+# MCU name
+MCU = STM32F303
+BOARD = QMK_PROTON_C
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
+# Build Options
+#   change yes to no to disable
```
Comment (fauxpark) [inline keyboards/handwired/twig/twig50/rules.mk:9]:
```suggestion
#   change yes to no to disable
#
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by nodatk** [comment]
@fauxpark 
Fixed again!

**Comment by nodatk** [comment]
What is the status of this PR?
If it has no more problem, please merge to the master.

File: keyboards/handwired/twig/twig50/README.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
+# Twig50
```
Comment (fauxpark) [inline keyboards/handwired/twig/twig50/README.md:1]: Please rename this file to lowercase.

File: keyboards/handwired/twig/twig50/twig50.h [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+	K10,      K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D, \
+	K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B,      K2D, \
+	K30, K31, K32,      K34, K35,      K37, K38, K39, K3A, K3B,      K3D  \
+) { \
+	{ K00, ___, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D }, \
+	{ K10, ___, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D }, \
+	{ K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, ___, K2D }, \
+	{ K30, K31, K32, ___, K34, K35, ___, K37, K38, K39, K3A, K3B, ___, K3D }  \
+}
+#define LAYOUT LAYOUT_diag_4x14
```
Comment (fauxpark) [inline keyboards/handwired/twig/twig50/twig50.h:21]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/twig/twig50/README.md (hunk 10 above)
Comment (nodatk) [reply keyboards/handwired/twig/twig50/README.md]: Renamed the file!

File: keyboards/handwired/twig/twig50/twig50.h (hunk 11 above)
Comment (nodatk) [reply keyboards/handwired/twig/twig50/twig50.h]: Fixed

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**